### ISSUE TITLE
Challenge completed with less than minimum-required amount of items

### ISSUE THREAD

**Issue by wellnesscookie** [BentoBoxWorld/Challenges#101]
### Description
- Challenge of ours that requires a minimum of 256x Arrow (Inventory Type) to be completed can be bypassed with any amount between x64 - x256, therefore if the player posseses only a stack of arrows, he can complete it while the required items are taken away from his inventory.

- It seems that this problem is only related to arrows, or at least they are the only items that we are aware of at the moment. We tried it for some blocks, but it seemed not to be bugging.

#### Steps to reproduce the behavior:
1. Have a challenge that requires x256 of arrows to be completed: https://prnt.sc/mwpkhb
2. Have at least a stack of arrows in the inventory
3. Try completing the challenge
4. ***Challenge completed***

#### Server / BentoBox information:
**Server information:**
- Database being used:
MySQL
- OS:
Debian GNU/Linux 9
- Java Version:
Java(TM) SE Runtime Environment 18.3 (build 10.0.1+10)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.1+10, mixed mode)

**BBox version output**:
```
[10:40:33] [main/INFO]: [CHAT] Running PAPER 1.13.2.
[10:40:33] [main/INFO]: [CHAT] BentoBox version: 1.4.0-SNAPSHOT
[10:40:33] [main/INFO]: [CHAT] Loaded Game Worlds:
[10:40:33] [main/INFO]: [CHAT] skyblock (skyblock)
[10:40:33] [main/INFO]: [CHAT] Loaded Add-Ons
[10:40:33] [main/INFO]: [CHAT] Biomes 0.5.0-SNAPSHOT
[10:40:33] [main/INFO]: [CHAT] BSkyBlock 1.3.0-SNAPSHOT
[10:40:33] [main/INFO]: [CHAT] Challenges 0.6.0
[10:40:33] [main/INFO]: [CHAT] Level 1.2.2-SNAPSHOT
[10:40:33] [main/INFO]: [CHAT] SerbCraftAddon 2.0
[10:40:33] [main/INFO]: [CHAT] WelcomeWarps 1.2.1-SNAPSHOT
```
**Plugins:**
`[06:32:17 INFO]: Plugins (50): AnimatedNames*, AreaShop, ArmorStandTools, BentoBox, BungeeTabListPlus*, ChatControl, ChestCommands, ChestShop*, ClearLag*, CrateReloaded, EditableSign, Essentials, EssentialsChat*, EssentialsSpawn*, ExecuteEverywhere*, FastAsyncWorldEdit*, FeatherBoard*, HeadDatabase*, HolographicDisplays, HungerKeeperPlus*, IPWhitelist*, IslandBorder, LeaderHeads*, LibsDisguises, LuckPerms, MobManager*, Multiverse-Core*, MVdWPlaceholderAPI*, NickRemover*, NoSleepCMDs*, OpenInv*, PlayerHeads, PlayerPoints*, ProtocolLib*, PvPManager*, PvPManagerBossBar*, RedstoneClockDetector*, RPGHealthIndicator*, SilkSpawners, SkinsRestorer, Spartan, TradeMe, TrophyHeads*, UnbreakingAnvils*, Vault*, VoidGenerator, WorldEdit, WorldGuard, WorldGuardPistonFix*, Yamler*`

#### Expected behavior / Additional context

- Have the challenge complete itself when a player posseses only a minimum required number of items (Arrows)

**Comment by BONNe** [comment]
Any errors in console?

**Comment by wellnesscookie** [comment]
Yes,

**When I tried to complete with x64 arrows**:
[11:50:38 ERROR]: [BentoBox] [Challenges] Could not remove 64 of ARROW from player's inventory!
[11:50:38 ERROR]: [BentoBox] [Challenges] Could not remove 64 of ARROW from player's inventory!
[11:50:38 ERROR]: [BentoBox] [Challenges] Could not remove 64 of ARROW from player's inventory!

**Found for the player that complained**:
https://prnt.sc/mwr2me

**Comment by BONNe** [comment]
Hmm, weird that it still proceed with completing a challenge.
I will check and fix it asap.

**Comment by BONNe** [comment]
Sorry that it took so long time.

I fixed this issue, but currently it will not return removed items. I still need to work on it.

**Comment by BONNe** [comment]
Implemented in PR #109
